src/Orleans.Runtime/Transactions/TransactionAgent.cs

```diff
@@ -274,11 +302,87 @@ public bool IsAborted(long transactionId)
 
         }
 
+        private async Task CommitTransactions(List<TransactionInfo> committingTransactions,
+            HashSet<long> outstandingCommits)
+        {
+            var stopWatch = Stopwatch.StartNew();
+            try
+            {
+                metrics.BatchCommitTransactionsRequestsCounter++;
+                var commitResponse = await this.tmService.CommitTransactions(committingTransactions, outstandingCommits);
+                metrics.BatchCommitTransactionsRequestLatencyCounter += stopWatch.Elapsed;
+                stopWatch.Stop();
```

Comment:
Probably doesn't matter, but usually call stop prior to getting the elapsed time.